### PR TITLE
Unify non-clustered and clustered interfaces

### PR DESCRIPTION
**Pull request by sazzad16**
An attempt to address [Common interface for <strike>binary</strike> clustered and non-clustered redis needed](https://github.com/xetorthio/jedis/issues/1545)

Inspiration: #1545 

Note: JDK 1.8 was required for using `default`.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -101,8 +101,8 @@
 				<artifactId>maven-compiler-plugin</artifactId>
 				<version>3.1</version>
 				<configuration>
-					<source>1.7</source>
-					<target>1.7</target>
+					<source>1.8</source>
```
Comment (gkorland) [inline pom.xml]: are we killing the 1.7 support?

File: src/main/java/redis/clients/jedis/Jedis.java [hunk 2]
```diff
@@ -3737,7 +3737,7 @@ public Long xlen(final String key) {
    * {@inheritDoc}
    */
   @Override
-  public long xack(final String key, final String group, final StreamEntryID... ids) {
+  public Long xack(final String key, final String group, final StreamEntryID... ids) {
```
Comment (gkorland) [inline src/main/java/redis/clients/jedis/Jedis.java]: I think it can cause some backward issues

File: pom.xml (hunk 1 above)
Comment (sazzad16) [reply pom.xml]: May be in future??

File: src/main/java/redis/clients/jedis/Jedis.java (hunk 2 above)
Comment (sazzad16) [reply src/main/java/redis/clients/jedis/Jedis.java]: That's why I've added this in major release. And it's necessary to maintain common/unified interface.

File: src/test/java/redis/clients/jedis/tests/commands/StreamsCommandsTest.java [hunk 3]
```diff
@@ -82,7 +82,7 @@ public void xdel() {
     assertEquals(2L, jedis.xlen("xdel-stream").longValue());
 
 
-    assertEquals(1L, jedis.xdel("xdel-stream", id1));
+    assertEquals(1L, jedis.xdel("xdel-stream", id1).longValue());
```
Comment (gkorland) [inline src/test/java/redis/clients/jedis/tests/commands/StreamsCommandsTest.java]: here is an example of backward...

**Comment by sazzad16** [comment]
@gkorland @marcosnils Please review!

@HeartSaVioR @mp911de Your comments would invaluable!

@greenrd @ptahchiev PTAL!

**Comment by mp911de** [comment]
It makes sense to expose a common API of base Redis commands for data structure and server interaction. Sentinel, Transactional, Cluster (CLUSTER SLOTS, CLUSTER MEET) and such commands are a separate issue. When working with Redis, the primary API concern should be working with the actual data structures regardless of the Redis operation mode. 

The operation mode is a separate aspect. A secondary concern is administrative API calls when using a Redis connector. A stand-alone API (e.g. new Jedis(...)) is good for regular data structure API calls as well as administrative calls. 

Therefore, I appreciate keeping segregated interfaces per Redis command group (how Redis documentation groups commands) and forming composite interfaces that are intended for use by applications (RedisCommands, RedisSentinelCommands, RedisStandaloneCommands, RedisClusterCommands). 

Does this help?

**Pull request closed without merging**